core/web/resolver/node_test.go

```diff
@@ -13,6 +14,93 @@ import (
 	"github.com/smartcontractkit/chainlink/core/utils"
 )
 
+func TestResolver_Nodes(t *testing.T) {
+	t.Parallel()
+
+	var (
+		chainID = *utils.NewBigI(1)
+		nodeID  = int32(200)
+
+		query = `
+			query GetNodes {
+				nodes {
+					results {
+						id
+						name
+						createdAt
+						chain {
+							id
+						}
+					}
+					metadata {
+						total
+					}
+				}
+			}`
+	)
+
+	testCases := []GQLTestCase{
+		unauthorizedTestCase(GQLTestCase{query: query}, "nodes"),
+		{
+			name:          "success",
+			authenticated: true,
+			before: func(f *gqlTestFramework) {
+				f.Mocks.evmORM.On("Nodes", PageDefaultOffset, PageDefaultLimit).Return([]types.Node{
+					{
+						ID:         nodeID,
+						Name:       "node-name",
+						EVMChainID: chainID,
+						CreatedAt:  f.Timestamp(),
+					},
+				}, 1, nil)
+				f.Mocks.evmORM.On("GetChainsByIDs", []utils.Big{chainID}).Return([]types.Chain{
+					{
+						ID: chainID,
+					},
+				}, nil)
+				f.App.On("EVMORM").Return(f.Mocks.evmORM)
+			},
+			query: query,
+			result: `
+			{
+				"nodes": {
+					"results": [{
+						"id": "200",
+						"name": "node-name",
+						"createdAt": "2021-01-01T00:00:00Z",
+						"chain": {
+							"id": "1"
+						}
+					}],
+					"metadata": {
+						"total": 1
+					}
+				}
+			}`,
+		},
+		{
+			name:          "generic error",
+			authenticated: true,
+			before: func(f *gqlTestFramework) {
+				f.Mocks.evmORM.On("Nodes", PageDefaultOffset, PageDefaultLimit).Return([]types.Node{}, 0, sql.ErrNoRows)
+				f.App.On("EVMORM").Return(f.Mocks.evmORM)
+			},
+			query:  query,
+			result: `null`,
+			errors: []*gqlerrors.QueryError{
+				{
+					Extensions:    nil,
+					ResolverError: sql.ErrNoRows,
```

Comment:
It's good to test this gql error but I think the test itself is misleading. A fetch of an array of results would not return a no rows error, because it would return an empty list.

Consider changing the test to a more general error